ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@lukewagner Encoding the argument list length in the operator encoding would solve your pre/post order concerns, but it adds an extra byte to all break operator encodings even those with one expression, and is an even worse encoding efficiency outcome than the current v8 encoding.

Further, I fail to see why the break operators should be the only places to construct a tuple/values. It's just as useful for the arguments to `if_else`, `select`, the `block` fall-through expression, etc. Why should we need `(block $l (br $l v1 v2 v3))` to construct a multi-value expression result - it just frustrates producers and is technically less efficient to encode.

Can anyone find fault with the br0/br proposal and with having separate tuple/values constructors in future for mv support? Fail-early-please.